test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -108,3 +115,53 @@ session_simulator::timestamp_transaction_uint(const std::string &ts_type, uint64
 
     return (0);
 }
+
+int
+session_simulator::query_timestamp(
+  const std::string &config, std::string &hex_ts, bool &ts_supported)
+{
+    std::string query_timestamp;
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+
+    /* For an empty config default to read timestamp. */
+    if (config.empty())
```

Comment:
I don't think we consider returning "read" timestamp if the config is empty. I checked "__txn_query_timestamp" function in src and it considers the empty as error.

Reply:
From the API documentation the config string is: _"a string, chosen from the following options: "commit", "first_commit", "prepare", "read"; default read."_ 

Also calling `self.session.query_timestamp()` with no arguments after setting the read timestamp in the python test produces this call log entry:

`{
    "class_name" : "session",
    "method_name" : "timestamp_transaction_uint",
    "session_id": "0x16bf880",
    "input" : {
        "wt_ts_txp_type": "read",
        "timestamp": 15
    },
    "output" : {
    },
    "return" : {
        "return_val" : 0,
        "error_message" : ""
    }
},
{
    "class_name" : "session",
    "method_name" : "query_timestamp",
    "session_id": "0x16bf880",
    "input" : {
        "config": "(null)"
    },
    "output" : {
        "timestamp_queried": "f"
    },
    "return" : {
        "return_val" : 0,
        "error_message" : ""
    }
},`

Reply:
In saying that I can't find where read gets set as the default config in the session API. There are also no python tests that call session.query_timestamp() with no config.

Reply:
As discussed I've removed this default behaviour so that query_timestamp() now returns an error if the config string doesn't include either commit, first_commit, prepare or read.